vllm/model_executor/layers/linear.py

```diff
@@ -280,6 +280,14 @@ def weight_loader(self,
                 if packed_dim == output_dim:
                     shard_size = shard_size // param.pack_factor
                     shard_offset = shard_offset // param.pack_factor
+
+                    # If marlin, we need to adjust the offset and size to account
```

Comment:
This part of the code snippets is repeated 4 times. May we consider writing it as a function?

Reply:
Thanks for pointing this out. Refactored the repeating code to a function.